src/main/java/Duke.java

```diff
@@ -29,9 +32,29 @@ public static int getTaskIndex(String userInput) {
         return taskIndex;
     }
 
-    public static void addList(Task[] list, int listCounter, String line) {
-        list[listCounter] = new Task(line);
-        System.out.println("added: " + line);
+    public static void addList(Task[] list, int listCounter, String userInput) {
+        String[] parsedUserInput = userInput.split(" ", 2);
+
+        if (parsedUserInput[0].equals(TODO_MESSAGE)) {
+            list[listCounter] = new ToDo(parsedUserInput[1]);
+        } else if (parsedUserInput[0].equals(DEADLINE_MESSAGE)) {
+            String[] deadlineInput = parsedUserInput[1].split("/by", 2);
```

Comment:
Likewise, you could consider renaming to plural form to show that it is an array

Reply:
yep thanks for spotting that too